modules/sdk-coin-sui/src/lib/transaction.ts

```diff
@@ -1,16 +1,354 @@
+    return this._suiTransaction;
+  }
+
+  setSuiTransaction(value: SuiTransaction): void {
+    this._suiTransaction = value;
+  }
+
+  /** @inheritDoc **/
+  get id(): string {
+    return this._id || UNAVAILABLE_TEXT;
+  }
+
+  getInputCoins(): SuiObjectRef[] {
+    return this.suiTransaction.kind.Single.Pay.coins;
+  }
+
+  /** @inheritdoc */
   canSign(key: BaseKey): boolean {
-    return false;
+    return true;
   }
 
+  /** @inheritdoc */
   toBroadcastFormat(): string {
-    throw new Error('Method not implemented.');
+    if (!this._suiTransaction) {
+      throw new InvalidTransactionError('Empty transaction');
+    }
+    return this.serialize();
   }
 
+  /** @inheritdoc */
   toJson(): TxData {
-    throw new Error('Method not implemented.');
+    if (!this._suiTransaction) {
+      throw new ParseTransactionError('Empty transaction');
+    }
+
+    return {
+      id: this._id,
+      gasBudget: this._suiTransaction.gasBudget,
+      gasPrice: this._suiTransaction.gasPrice,
+      kind: this._suiTransaction.kind,
+      sender: this._suiTransaction.sender,
+      gasPayment: this._suiTransaction.gasPayment,
+    };
+  }
+
+  /** @inheritDoc */
+  explainTransaction(): TransactionExplanation {
+    const result = this.toJson();
+    const displayOrder = ['id', 'outputs', 'outputAmount', 'changeOutputs', 'changeAmount', 'fee', 'type'];
+    const outputs: TransactionRecipient[] = [];
+
+    const explanationResult: TransactionExplanation = {
+      displayOrder,
+      id: this.id,
+      outputs,
+      outputAmount: '0',
+      changeOutputs: [],
+      changeAmount: '0',
+      fee: { fee: this.suiTransaction.gasBudget.toString() },
+      type: this.type,
+    };
+
+    switch (this.type) {
+      case TransactionType.Send:
+        return this.explainTransferTransaction(result, explanationResult);
+      default:
+        throw new InvalidTransactionError('Transaction type not supported');
+    }
+  }
+
+  /**
+   * Set the transaction type.
+   *
+   * @param {TransactionType} transactionType The transaction type to be set.
+   */
+  transactionType(transactionType: TransactionType): void {
+    this._type = transactionType;
+  }
+
+  /**
+   * Load the input and output data on this transaction.
+   */
+  loadInputsAndOutputs(): void {
+    if (!this.suiTransaction) {
+      return;
+    }
+
+    const tx = this.suiTransaction;
+    const payTx = tx.kind.Single.Pay;
+    const recipients = payTx.recipients;
+    const amounts = payTx.amounts;
+    if (recipients.length !== amounts.length) {
+      throw new Error('The length of recipients does not equal to the length of amounts');
+    }
+
+    const outputs: Entry[] = recipients.map((recipient, index) => ({
+      address: recipient,
+      value: amounts[index].toString(),
+      coin: this._coinConfig.name,
+    }));
+
+    const totalAmount = amounts.reduce((accumulator, current) => accumulator + current, 0);
+    const inputs: Entry[] = payTx.coins.map((input) => ({
```

Comment:
i think we should only has one input, because we don't know each coin's balance.

Reply:
make sense, updated!